src/protocol/sort/bit_permutation.rs

```diff
@@ -3,83 +3,90 @@ use std::iter::{repeat, zip};
 use crate::{
     error::BoxError,
     ff::Field,
-    protocol::{context::ProtocolContext, RecordId},
-    secret_sharing::Replicated,
+    protocol::{context::ProtocolContext, RecordId, RECORD_0},
+    secret_sharing::{MaliciousReplicated, Replicated, SecretSharing},
 };
 
 use crate::protocol::mul::SecureMul;
 use embed_doc_image::embed_doc_image;
 use futures::future::try_join_all;
 
-/// This is an implementation of `GenBitPerm` (Algorithm 3) described in:
-/// "An Efficient Secure Three-Party Sorting Protocol with an Honest Majority"
-/// by K. Chida, K. Hamada, D. Ikarashi, R. Kikuchi, N. Kiribuchi, and B. Pinkas
-/// <https://eprint.iacr.org/2019/695.pdf>.
-#[derive(Debug)]
-pub struct BitPermutation<'a, F: Field> {
-    input: &'a [Replicated<F>],
+pub trait ShareOfOne<F: Field> {
+    type Share: SecretSharing<F>;
+    fn share_of_one(&self) -> Self::Share;
 }
 
-impl<'a, F: Field> BitPermutation<'a, F> {
-    pub fn new(input: &'a [Replicated<F>]) -> BitPermutation<'a, F> {
-        Self { input }
+impl<F: Field> ShareOfOne<F> for ProtocolContext<'_, Replicated<F>, F> {
+    type Share = Replicated<F>;
+
+    fn share_of_one(&self) -> Self::Share {
+        Replicated::one(self.role())
     }
+}
```

Comment:
Stuff like this should not be defined within this file `bit_permutation.rs` - this is a particular protocol, but many protocols will require a share of one.

Reply:
Created a new file called context_trail.rs and moving this code there. Other context related trait definitions can also be added here.